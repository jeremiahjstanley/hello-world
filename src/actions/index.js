export const setLocation = (location) => ({
  type: 'SET_LOCATION',
  location
});

export const setDataSet = (dataSet) => ({
  type: 'SET_DATASET',
  dataSet
});

export const clearDataSet = () => ({
  type: 'CLEAR_DATASET',
});

export const setDataBase = (dataBase) => ({
  type: 'SET_DATABASE',
  dataBase
});

export const fetchLocationDataSuccess = (locationData) => ({
  type: 'FETCH_LOCATION_DATA_SUCCESS',
  locationData
});

export const hasErrored = (status) => ({
  type: 'HAS_ERRORED',
  hasErrored: status
});

export const madeSearch = (status) => ({
  type: 'MADE_SEARCH',
  madeSearch: status
});


