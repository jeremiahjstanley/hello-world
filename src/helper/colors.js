export const colors = [
'#e700ff',
'#0081ff',
'#ff0063',
'#74d2e7',
'#00f1ff',
'#fd5c63',
'#48a9c5',
'#0085ad',
'#8db9ca',
'#4298b5',
'#00205b',
'#d20962',
'#f47721',
'#00a78e',
'#7d3f98',
'#037ef3',
'#279b37',
'#008374',
'#0099e5',
'#ff4c4c',
'#34bf49'
]