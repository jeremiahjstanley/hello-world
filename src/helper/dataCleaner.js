export const goveranceIndicatorCleaner = (estimates, numberOfSources, percentileRank, standardError) => {
  const cleanEstimates = estimates.dataset.data.map(dataPoint => {
    return {"name": dataPoint[0], "Estimate": dataPoint[1]}
  });
  const cleanNumberOfSources = numberOfSources.dataset.data.map((dataPoint, index) => {
    return {...cleanEstimates[index], "name": dataPoint[0], "Number of Sources": dataPoint[1]}
  })
  const cleanPercentileRank = percentileRank.dataset.data.map((dataPoint, index) => {
    return {...cleanNumberOfSources[index], "name": dataPoint[0], "Percentile Rank": dataPoint[1]}
  })
  const cleanStandardError = standardError.dataset.data.map((dataPoint, index) => {
    return {...cleanPercentileRank[index], "name": dataPoint[0], "Standard Error": dataPoint[1]}
  })
  return cleanStandardError
}