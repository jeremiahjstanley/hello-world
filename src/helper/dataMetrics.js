export const dataMetrics = [
  {
    name: 'Voice and Accountability', 
    dataset_code: 'VA', 
    database_code: 'WWGI'
  },
  {
    name: 'Political Stability and Absence of Violence',
    dataset_code: 'PV',
    database_code: 'WWGI'
  },
  {
    name: 'Government Effectiveness',
    dataset_code: 'GE',
    database_code: 'WWGI'},
  {
    name: 'Regulatory Quality',
    dataset_code: 'RQ',
    database_code: 'WWGI'
  },
  {
    name: 'Rule of Law',
    dataset_code: 'RL',
    database_code: 'WWGI'
  },
  {
    name: 'Control of Corruption', 
    dataset_code: 'CC',
    database_code: 'WWGI'},
]
