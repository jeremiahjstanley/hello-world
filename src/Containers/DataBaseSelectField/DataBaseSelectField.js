import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setDataBase } from '../../actions';
import { dataMetrics } from '../../helper/dataMetrics';

export class DataBaseSelectField extends Component {
  constructor() {
    super()

    this.state = {
      dataBase: {name: dataMetrics[0].name, database_code: dataMetrics[0].database_code}
    }
  }

  componentDidMount() {
    this.props.selectDataBase(this.state.dataBase)
  }

  selectMetric = (event) => {
    const dataBase = dataMetrics.find(metric => event.target.value === metric.name);
    const dataStore = {name: dataBase.name, database_code: dataBase.database_code}
    this.props.selectDataBase(dataStore)
  }

  render() {

    const options = dataMetrics.map((metric, index) => (
      <option key={index}>
        { metric.name }
      </option>
    ))

    return (
      <select onChange={this.selectMetric}>
        { options }
      </select>
    )
  };
}

export const mapDispatchToProps = (dispatch) => ({
  selectDataBase: (dataBase) => dispatch(setDataBase(dataBase))
})

export default connect(null, mapDispatchToProps)(DataBaseSelectField);