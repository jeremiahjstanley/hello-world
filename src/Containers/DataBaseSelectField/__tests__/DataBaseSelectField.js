import React from 'react';
import { shallow } from 'enzyme';
import { DataBaseSelectField, mapDispatchToProps } from '../DataBaseSelectField';
import { setDataBase } from '../../../actions';
import { dataMetrics } from '../../../helper/dataMetrics';


describe('DataBaseSelectField', () => {

  let wrapper;
  let mockEvent;
  let mockSelectDataBase;
  let mockDataBase;

  beforeEach(() => {

    mockDataBase = {
  		name: 'Worldwide Governance Indicators',
  		database_code: 'WWGI'
  	};

  	mockSelectDataBase = jest.fn();

    wrapper = shallow(
    	<DataBaseSelectField 
    		selectDataBase={mockSelectDataBase}
    	/>
  	);

  });

	it('should match the snapshot', () => {

		expect(wrapper).toMatchSnapshot();

	});

  it('should call selectDataBase when the dropdown is has a change event', () => {

  	mockEvent = { target: { value: 'Worldwide Governance Indicators'} };

    wrapper.instance().selectMetric(mockEvent);

    expect(mockSelectDataBase).toHaveBeenCalledWith(mockDataBase);

  });

	describe('mapDispatchToProps', () => {

		let actionToDispatch;
		let mockDispatch;

		beforeEach(() => {

			mockDispatch = jest.fn();

		});

		it('calls dispatch with the setDataBase action when called', () => {

			const mappedProps = mapDispatchToProps(mockDispatch);

			actionToDispatch = setDataBase(mockDataBase);

			mappedProps.selectDataBase(mockDataBase);

			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);

		});
		
	});
	
});