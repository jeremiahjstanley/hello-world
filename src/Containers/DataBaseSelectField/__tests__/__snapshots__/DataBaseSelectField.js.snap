// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataBaseSelectField should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataBaseSelectField
    selectDataBase={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option>
          Worldwide Governance Indicators
</option>,
        <option>
          Worldwide Poverty Statistics
</option>,
        <option>
          Worldwide Gender Statistics
</option>,
        <option>
          Worldwide Education Statistics
</option>,
        <option>
          Worldwide Health Nutrition and Population Statistics
</option>,
        <option>
          Worldwide Development Indicators
</option>,
      ],
      "onChange": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Worldwide Governance Indicators",
        },
        "ref": null,
        "rendered": "Worldwide Governance Indicators",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Worldwide Poverty Statistics",
        },
        "ref": null,
        "rendered": "Worldwide Poverty Statistics",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Worldwide Gender Statistics",
        },
        "ref": null,
        "rendered": "Worldwide Gender Statistics",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "Worldwide Education Statistics",
        },
        "ref": null,
        "rendered": "Worldwide Education Statistics",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "Worldwide Health Nutrition and Population Statistics",
        },
        "ref": null,
        "rendered": "Worldwide Health Nutrition and Population Statistics",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": "Worldwide Development Indicators",
        },
        "ref": null,
        "rendered": "Worldwide Development Indicators",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Worldwide Governance Indicators
</option>,
          <option>
            Worldwide Poverty Statistics
</option>,
          <option>
            Worldwide Gender Statistics
</option>,
          <option>
            Worldwide Education Statistics
</option>,
          <option>
            Worldwide Health Nutrition and Population Statistics
</option>,
          <option>
            Worldwide Development Indicators
</option>,
        ],
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Worldwide Governance Indicators",
          },
          "ref": null,
          "rendered": "Worldwide Governance Indicators",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Worldwide Poverty Statistics",
          },
          "ref": null,
          "rendered": "Worldwide Poverty Statistics",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Worldwide Gender Statistics",
          },
          "ref": null,
          "rendered": "Worldwide Gender Statistics",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "Worldwide Education Statistics",
          },
          "ref": null,
          "rendered": "Worldwide Education Statistics",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "Worldwide Health Nutrition and Population Statistics",
          },
          "ref": null,
          "rendered": "Worldwide Health Nutrition and Population Statistics",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": "Worldwide Development Indicators",
          },
          "ref": null,
          "rendered": "Worldwide Development Indicators",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
